app/src/lib/git/diff.ts

```diff
@@ -105,12 +113,14 @@ export async function getCommitDiff(
     args.push(file.oldPath)
   }
 
-  const { output } = await spawnAndComplete(
+  const { output, didReadAllBytes } = await spawnAndComplete(
     args,
     repository.path,
-    'getCommitDiff'
+    'getCommitDiff',
+    undefined,
+    MaxBytesToRead
   )
-  if (isBufferTooLarge(output)) {
+  if (!didReadAllBytes || isBufferTooLarge(output)) {
```

Comment:
Following on from my comment about using `MaxReasonableDiffSize`, could we use `output.length >= MaxReasonableDiffSize` here and avoid introducing a property on the return type here? We could extract that logic into a method similar to `isBufferToLarge` i.e. `isBufferReasonable` and then once we get to the point where we want to be able to tell this method whether to load the diff anyway we can elect to not care about the soft limit.

Reply:
We're doing that already with `isBufferTooLarge`. Getting the length of the buffer will not tell us that `spawnAndComplete` exited before reading the entire buffer, so we need that property. The alternative is to pass a callback to `spawnAndComplete` that get's fired if it doesn't complete reading a buffer.

Reply:
> Getting the length of the buffer will not tell us that spawnAndComplete exited before reading the entire buffer

Do we care though? If the length is greater than or equal to the `MaxReasonableDiffSize` we shouldn't be displaying the diff unless the user explicitly asks us to. I might be missing something but it seems to me the only case where we couldn't determine whether `spawnAndComplete` exited before reading all of it is when the output is exactly as large as `MaxReasonableDiffSize` and at that point does it even matter?